sale_coupon_multi_gift/models/sale_coupon_reward.py

```diff
@@ -38,11 +41,29 @@ class SaleCouponGift(models.Model):
     reward_product_quantity = fields.Integer(
         string="Quantity", help="Reward product quantity",
     )
-    reward_product_id = fields.Many2one(
-        comodel_name="product.product", string="Free Product", help="Reward Product",
+    reward_default_product_id = fields.Many2one(
+        comodel_name="product.product",
+        compute="_compute_reward_default_product_id",
+        inverse="_inverse_reward_default_product_id",
+        readonly=False,
     )
-    reward_product_uom_id = fields.Many2one(
-        related="reward_product_id.product_tmpl_id.uom_id",
-        string="Unit of Measure",
-        readonly=True,
+    reward_product_ids = fields.Many2many(
+        comodel_name="product.product",
+        string="Free Product Options",
+        help="Reward Product",
     )
+
+    @api.depends("reward_product_ids")
+    def _compute_reward_default_product_id(self):
+        """This field acts as a cover for a simple many2one behavior of the module.
+        Another module like `sale_coupon_selection_wizard` can dismiss it in order
+        to allow optional"""
+        for line in self:
+            if line.reward_product_ids:
+                line.reward_default_product_id = fields.first(line.reward_product_ids)
+            else:
+                line.reward_default_product_id = False
```

Comment:
AFAIK, this works the same:

```suggestion
            line.reward_default_product_id = fields.first(line.reward_product_ids)
```

Reply:
Nice :+1: